### ISSUE TITLE
Revisit assert/verify/enforce separation

### ISSUE THREAD

**Issue by mihails-strasuns-sociomantic** [sociomantic-tsunami/ocean#289]
https://github.com/sociomantic-tsunami/ocean/issues/212 resulted in a very straightforward migration where all `assert` places got changed to `verify`/`test` without any detailed consideration. This unblocks further D2 build deployment but in the long term we would need something finer tuned.

We should revisit all modules either one by one or by package and check what can be changed back to `assert` and what should really be an `enforce` with proper exception type. Eventually `-release` should be able to remove all non-essential checks.

**Comment by mathias-lang-sociomantic** [comment]
>  This unblocks further D2 build deployment but in the long term we would need something finer tuned.

Does it ? For the dhtnode, don't we need to do the same in swarm and dhtproto ?

**Comment by gavin-norman-sociomantic** [comment]
Yes, we need to do the same process in the other libraries. From ocean's side, though, things are complete.

**Comment by mihails-strasuns-sociomantic** [comment]
Right, I only speak in context of ocean as it is ocean issue :)

**Comment by leandro-lucarella-sociomantic** [comment]
Maybe a global issue in *backend* could help track the global progress (or a GitHub project).

**Comment by mihails-strasuns-sociomantic** [comment]
That is just a bit too globa :). Ocean alone will take uncertain amount of time
